[FEATURE] Redesigned Settings-Screen and Workflow

chfsx opened the pull request:
This branch contains the completely revised UI for the PC plugin in the settings view and also in the rendering of the content. See screenshot.

![image](https://github.com/user-attachments/assets/4179e7b8-e77b-45aa-9e23-743ab580e535)


To discuss:
Do we need the aspect ratio? I have now implemented it so that you can adopt the aspect ratio of the publication by default. However, if desired, you can overwrite this.

This branch requires the following branch of the main plugin: https://github.com/opencast-ilias/OpenCast/pull/430

chfsx commented:
This is what I like about such refactorings: 
<img width="150" alt="image" src="https://github.com/user-attachments/assets/f857f2fe-58ac-49ad-9353-af8534cc6dc3" />

dagraf commented:
The initial issue described here (https://github.com/opencast-ilias/OpencastPageComponent/issues/35#issue-2743413625) appears with this PR.

dagraf commented:
Titel "Bestehendes Video aus Opencast auswählen" is missing (see screenshot from Mockup).

<img width="578" alt="Bildschirmfoto 2025-04-22 um 21 42 46" src="https://github.com/user-attachments/assets/a665f397-d153-4bcb-bf8c-af5a388771b7" />

dagraf commented:
Please write title "Insert existing video from Opencast" NOT bold (same for all other titles, such as "Aktuell ausgewähltes Video" and "Eigenschaften") and remove subtitle "Events" (see screenshot).

<img width="1167" alt="Bildschirmfoto 2025-04-22 um 21 45 57" src="https://github.com/user-attachments/assets/a23842e2-82db-4ab7-84ab-42527f5007b9" />

dagraf commented:
EDIT (16.5.): Fixed and approved.

When setting "Seitenverhältnis" is set to 16:9, the inserted video is not displayed in 16:9 format (see screenshots). When it is set to "As in publication" and the format of the video is 16:9, the video gets correctly displayed in 16:9 format.

<img width="848" alt="Bildschirmfoto 2025-04-22 um 21 55 04" src="https://github.com/user-attachments/assets/8aeb98ec-fa1b-4c80-8471-b8b4ea8f3132" />
<img width="980" alt="Bildschirmfoto 2025-04-22 um 21 55 31" src="https://github.com/user-attachments/assets/dafa7965-8693-440b-985e-c21f547bc5c5" />

dagraf commented:
EDIT (16.5.2025): Fixed and approved.

A lot of English translations are missing (see screenshot for example).

<img width="853" alt="Bildschirmfoto 2025-04-22 um 22 00 17" src="https://github.com/user-attachments/assets/f733037f-1a60-4c6b-8019-5f710b1d4fa3" />

dagraf commented:
Explanation of option "Seitenverhältnis" is not as specified (see screenshot of mockup).

<img width="519" alt="Bildschirmfoto 2025-04-22 um 22 01 15" src="https://github.com/user-attachments/assets/99164a21-fced-4806-9e95-41fba7129239" />

dagraf commented:
EDIT (16.5.): Fixed and approved.

When choosing an Opencast video for the first time, is the actual width of the video taken as "Maximale Breite" or the configured "Default width" (see config of plugin)? In my tests, it seems like the configured "Default width" gets shown.

dagraf commented:
Specified "Hover-Over Info" ("Opencast-Video") is missing (see screenshot of mockup).

<img width="387" alt="Bildschirmfoto 2025-04-22 um 22 08 26" src="https://github.com/user-attachments/assets/1ad1a2a3-28fb-47d7-a369-cd25bb8e2556" />

dagraf commented:
Correct English version of "Bestehendes Video aus Opencast auswählen" (see screenshot: "Insert existing video from Opencast") to "Select existing video from Opencast". While doing so, please add missing English translations of "Auswählen" (see screenshot, English: "Select").

<img width="1167" alt="Bildschirmfoto 2025-04-22 um 22 13 04" src="https://github.com/user-attachments/assets/fcc800c9-d61b-419f-be99-53e1595d7aeb" />

dagraf commented:
When an admin does "Ersetzen" an already inserted Opencast video, the success message "Video successfully selected." (or "... succesfully exchanged.") is missing.

dagraf commented:
Ordering issues:
- Ordering Z-A for "Title" does not work correctly. A-Z works as expected.
- Orderin for "Series" does not work at all. When hitting on the the row title "Series" a empty table is shown (see screenshot).
 
<img width="1175" alt="Bildschirmfoto 2025-04-22 um 22 21 28" src="https://github.com/user-attachments/assets/3a13c10e-e430-4729-a456-fb231bbd5289" />

dagraf commented:
Option "Vorschaubild verlinkt auf Video" does not work. When activated, a click on the shown thumbnail has no effect.

dagraf commented:
EDIT (16.5.): Fixed and approved.

Inserted videos sometimes show an error. I can not recognize any rule here. But here's what I observe:
- I have inserted 5 Opeoncast videos.
- When I reload my browser, always different videos show issues (except the one "Vorschaubild verlinkt auf Video" at the very top). See screenshots.
- There seem to be 2 different type of errors: (i) SQLSTATE[23000]: Integrity constraint violation: ...; (ii) Trying to access array offset on value of type null.

I will talk to @snoesberger if this could be an issue with our test environment.

<img width="448" alt="Bildschirmfoto 2025-04-22 um 22 27 44" src="https://github.com/user-attachments/assets/4bca6126-9640-445e-b53b-512aaddb8ade" />
<img width="395" alt="Bildschirmfoto 2025-04-22 um 22 27 52" src="https://github.com/user-attachments/assets/6abdbed0-67ac-467b-ae43-388bd56b0098" />
<img width="422" alt="Bildschirmfoto 2025-04-22 um 22 28 03" src="https://github.com/user-attachments/assets/dd589706-4edf-4062-b68c-131015d4a0e5" />
<img width="408" alt="Bildschirmfoto 2025-04-22 um 22 28 15" src="https://github.com/user-attachments/assets/38715d1b-484c-4e5d-956b-0c646d226f93" />

snoesberger commented:
> Inserted videos sometimes show an error. I can not recognize any rule here. But here's what I observe:
> 
>     * I have inserted 5 Opeoncast videos.
> 
>     * When I reload my browser, always different videos show issues (except the one "Vorschaubild verlinkt auf Video" at the very top). See screenshots.
> 
>     * There seem to be 2 different type of errors: (i) SQLSTATE[23000]: Integrity constraint violation: ...; (ii) Trying to access array offset on value of type null.
> 
> 
> I will talk to @snoesberger if this could be an issue with our test environment.

I've checked our Opencast test environment and everything is fine, I can't see any errors in the Opencast logs either. So I assume the problem is on the plugin side.

chfsx commented:
@dagraf I try to write your findings in a list with checkboxes, so we have a better overview. I will update this comment with feedback and update the checkboxes for resolved issues.

- [x] Please write title "Insert existing video from Opencast" NOT bold (same for all other titles, such as "Aktuell ausgewähltes Video" and "Eigenschaften") and remove subtitle "Events" (see screenshot).

> We do not influence anything ourselves (e.g. bold etc.), that is how the UI components look and we cannot/should not influence that. the subheading, for example, is required by the table, so we have to write something in there. of course, we could also insert an empty string ‘’, but that actually contradicts the rules of the UI components. 

- [x] When setting "Seitenverhältnis" is set to 16:9, the inserted video is not displayed in 16:9 format (see screenshots). When it is set to "As in publication" and the format of the video is 16:9, the video gets correctly displayed in 16:9 format. > Fix confirmed by @dagraf 

> I managed to fix that, will be part of the next deployment (I will inform you as soon as the ReviewApp is updated)

- [x] A lot of English translations are missing (see screenshot for example). > Fix confirmed by @dagraf 

> I added several missing translations now

- [x] Explanation of option "Seitenverhältnis" is not as specified (see screenshot of mockup). > Fix confirmed by @dagraf

> @dagraf in the mockup the option is labled/described with "Seitenverhältnis des ausgewählten Videos. Wird so in die Seite eingefügt.". Is there another ver4sion of the mockups?
>> @chfsx: No, there is not. But this description is not very well chosen. Please use instead: (EN: The video is inserted with this aspect ratio. / DE: Das Video wird mit diesem Seitenverhältnis eingefügt.)
>>> I updated the translation now

- [x] When choosing an Opencast video for the first time, is the actual width of the video taken as "Maximale Breite" or the configured "Default width" (see config of plugin)? In my tests, it seems like the configured "Default width" gets shown. > Fix confirmed by @dagraf 

> This is fixed now and the width of the original publication is used if possible

- [x] Specified "Hover-Over Info" ("Opencast-Video") is missing (see screenshot of mockup).

> I haven't found a way for a plugin to influence this. In my opinion, this is a bug in the core that prevents the names of the plugins from being displayed here. I have also tried this with other PageComponentPlugins, but none of them display anything here.
>> @chfsx: Would a bug report in the core make sense here? If so, can you please create one?
>>> I reported this here: https://mantis.ilias.de/view.php?id=45215


- [x] Correct English version of "Bestehendes Video aus Opencast auswählen" (see screenshot: "Insert existing video from Opencast") to "Select existing video from Opencast". While doing so, please add missing English translations of "Auswählen" (see screenshot, English: "Select"). > Fix confirmed by @dagraf

> I updated the english language file
>> @chfsx: In the table, it still says "Auswählen". Correct would be "Select".
>>> The english language file is now updated correctly

- [x] When an admin does "Ersetzen" an already inserted Opencast video, the success message "Video successfully selected." (or "... succesfully exchanged.") is missing. > Fix confirmed by @dagraf

> Done

- [ ] Ordering Z-A for "Title" does not work correctly. A-Z works as expected.

> @dagraf this seems to be related to this issue in core: https://mantis.ilias.de/view.php?id=45177

- [x] Orderin for "Series" does not work at all. When hitting on the the row title "Series" a empty table is shown (see screenshot). > Bug not reproducable also by @dagraf

> @dagraf I cannot reproduce this, is this still a thing?

- [x] Option "Vorschaubild verlinkt auf Video" does not work. When activated, a click on the shown thumbnail has no effect. > Fix confirmed by @dagraf

> Done

- [x] Inserted videos sometimes show an error. I can not recognize any rule here. > Fix confirmed by @dagraf 

> Done

The ReviewApp is updated with the checked issues: http://ocpc-pr-38.opencast.k8s.sr.solutions/

chfsx commented:
@dagraf The ReviewApp is updated with the checked issues: http://ocpc-pr-38.opencast.k8s.sr.solutions/

chfsx commented:
Deployment is running https://github.com/opencast-ilias/ReviewApps/actions/runs/15414329854

dagraf commented:
All tests of bugfixes were successfull. Therefore and @ferishili, can you please review this PR? Thx!

dagraf commented:
Filter do work. I was able to test them successfull.

But: 

- [x] The amount of shown rows is not correct (see original issue https://github.com/opencast-ilias/OpencastPageComponent/issues/35#). See screenshot below. Not published videos do not get shown. The "simple variant" we decided to go with is still ok.

<img width="1180" alt="Bildschirmfoto 2025-06-03 um 17 24 59" src="https://github.com/user-attachments/assets/04be4bd1-7bbe-4a57-bfb2-e41cee9308b0" />

chfsx commented:
@dagraf the ReviewApp has been updated (Workflow running here: https://github.com/opencast-ilias/ReviewApps/actions/runs/15674593870) the fix the issue concerning amount of items in table

chfsx commented:
The workflow has finished and the ReviewApp is ready: http://ocpc-pr-38.opencast.k8s.sr.solutions/ Please note that this is an older ReviewApp with only the "root" account

dagraf commented:
I can confirm the bugfix. Therefore and @ferishili: Can you please review this PR? Thx!

ferishili commented on lang/ilias_en.lang:
```diff
@@ -6,18 +6,27 @@ form_title#:#Edit Video Properties
+max_width#:#Maximum width
+max_width_info#:#Optional: Set the maximum width of the video (in pixels). The effective width of the video is pre-filled.
+link_thumbnail#:#Preview image linked to video
+link_thumbnail_info#:#If active, the video will open in a new tab when the preview image is clicked. Otherwise, the video will be embedded in the page and clicking on it will start the video directly.
```
Adjustments for this description:

```suggestion
link_thumbnail_info#:#When enabled, clicking the preview image will open the video in a new browser tab.  
Otherwise, the video will be embedded directly on the page and start playing when clicked.
```

dagraf replied:
I like these adjustments. Thx!

ferishili commented on lang/ilias_de.lang:
```diff
@@ -6,18 +6,27 @@ form_title#:#Videoeigenschaften Bearbeiten
+max_width#:#Maximale Breite
+max_width_info#:#Optional: Stellen Sie die maximale Breite des Videos ein (in Pixeln). Vorausgefüllt ist die effektive Breite des Videos.
+link_thumbnail#:#Vorschaubild verlinkt auf Video
+link_thumbnail_info#:#Wenn aktiv, wird das Video bei einem Klick auf das Vorschaubild in einem neuen Tab geöffnet. Ansonsten wird das Video in der Seite eingebunden und ein Klick darauf startet das Video direkt.
```
An adjustment here:

```suggestion
link_thumbnail_info#:#Wenn aktiviert, wird das Video bei einem Klick auf das Vorschaubild in einem neuen Browser-Tab geöffnet. Ansonsten wird das Video in der Seite eingebunden und ein Klick darauf startet das Video direkt.
```

dagraf replied:
I like these adjustments. Thx!

ferishili commented on src/Authorization/TokenAR.php:
```diff
@@ -133,8 +149,6 @@ public function sleep($field_name)
      * @param $field_name
      * @param $field_value
-     *
-     * @return Token|null
```
Is there a reason you removed the `@return`?

ferishili replied:
What about here? @chfsx

ferishili replied:
?

ferishili commented on src/Views/Dimensions.php:
```diff
@@ -0,0 +1,48 @@
+
+    public function getRatio(): float
+    {
+        return (float) $this->height / $this->width;
```
Is the `(float)` meant to cast the `$this->height` only?
If it is meant to cast the result of division, it needs to be wrapped in a paranthesis, like `(float) ($this->height / $this->width)`

ferishili commented on src/Views/Display.php:
```diff
@@ -0,0 +1,178 @@
+        $max_width = (string) ($this->properties[\ilOpencastPageComponentPluginGUI::PROP_WIDTH] ?? '');
+        $max_width = $max_width !== '' ? $max_width . 'px' : 'auto';
+
+        $configured_aspect_ratio = (float) ($this->properties[\ilOpencastPageComponentPluginGUI::PROP_ASPECT_RATIO] ?? Edit::RATIO_AS_PUBLICATION);
```
`Edit::RATIO_AS_PUBLICATION` as fallback and then casting to `float` would make the `===` comparison in next line alway `false`, therefore the (float) casting must be limited to the `$this->properties[\ilOpencastPageComponentPluginGUI::PROP_ASPECT_RATIO]`

```suggestion
        $configured_aspect_ratio =  ((float) $this->properties[\ilOpencastPageComponentPluginGUI::PROP_ASPECT_RATIO]) ?? Edit::RATIO_AS_PUBLICATION);
```

ferishili commented on src/Views/Display.php:
```diff
@@ -0,0 +1,178 @@
+
+        $content = match (true) {
+            $this->event === null => $this->getExceptionHTML($this->properties),
+            $as_link || $this->mode !== \ilOpencastPageComponentPluginGUI::MODE_PRESENTATION => $this->getStandardElementHTML(
```
Please wrap the `$this->mode !== ...` comparison in parentheses to improve readability.

ferishili replied:
What about here? @chfsx

ferishili commented on src/Views/Display.php:
```diff
@@ -0,0 +1,178 @@
+        );
+    }
+
+    // Moved from old class
```
Is this needed here?

ferishili replied:
What about here? @chfsx

ferishili commented on src/Views/Edit.php:
```diff
@@ -0,0 +1,161 @@
+        // get image size
+        $size = getimagesizefromstring($thumbnail);
+        // calculate aspect ratio
+        $aspect_ratio = $size[0] / $size[1];
```
how about casting the whole division to float here?
```suggestion
        $aspect_ratio = (float) ($size[0] / $size[1]);
```

ferishili replied:
What about here? @chfsx

ferishili commented on src/Views/EventDimensions.php:
The method `determineForEvent` has 3 returns, which could be reduced to one with use of a default `$ratio = null;` and give it the value based on the condition and return it at the end!

ferishili replied:
What about here? @chfsx

ferishili replied:
This was not addressed, but it is functional without any problem, therefore, in favor of speeding up the process I would skip this!
Just keep in mind that there is room for improvments here!

ferishili commented on templates/css/presentation.css:
```diff
@@ -1,73 +1,59 @@
+    grid-column: 1;
+    display: flex;
+    background-color: rgba(0, 0, 0, 0.6);
+    /*background: rgba(2, 0, 36, 7.16);*/
```
should we keep these 2 lines?

ferishili replied:
What about here? @chfsx

ferishili commented on templates/css/presentation.css:
```diff
@@ -1,73 +1,59 @@
+    width: 48px;
+    height: 48px;
+    margin: auto;
+    /*opacity: 0.6;*/
```
should we keep it?

ferishili replied:
What about here? @chfsx

ferishili commented on templates/html/component_as_link.html:
```diff
@@ -1,17 +1,8 @@
+<a class="ocpc-video-link" href="{VIDEO_LINK}" target="{TARGET}" {ADDITIONAL_LINK_ATTRIBUTES} {MODAL_LINK}>
+    <img class="ocpc-thumbnail" src="{THUMBNAIL_URL}">
+    <div class="ocpc-play-overlay">
+        <img class="ocpc-play-icon" src="./Customizing/global/plugins/Services/COPage/PageComponent/OpencastPageComponent/templates/images/play.svg">
```
Does this relative path as for `src="./Customizing/global/plugins/Services/COPage/PageComponent/OpencastPageComponent/templates/images/play.svg"` work properly? Considering it will be rendered and read the path from the ILIAS root!

ferishili replied:
What about here? @chfsx

chfsx replied:
This should work IMO, this was used the same way before

ferishili commented on templates/html/tpl.container.html:
```diff
@@ -0,0 +1,3 @@
+<div class="ocpc-container" style="max-width: {MAX_WIDTH}; aspect-ratio: {RATIO}">
```
missing a semicolon!

```suggestion
<div class="ocpc-container" style="max-width: {MAX_WIDTH}; aspect-ratio: {RATIO};">
```

ferishili commented on classes/class.ilOpencastPageComponentPluginGUI.php:
```diff
@@ -133,166 +155,71 @@ public function performCommand(string $cmd): void
-        $upload_button->setCaption($this->plugin->txt('btn_upload'), false);
-        $this->dic->toolbar()->addButtonInstance($upload_button);
-    }
+    // we want to reducet the code here, therefore we delegate some thing to other classes (TODO)
```
Either remove this comment or write format the TODO properly:

```suggestion
    // TODO: we want to reducet the code here, therefore we delegate some thing to other classes.
```

chfsx replied:
I will remove the line

ferishili commented on classes/class.ilOpencastPageComponentPluginGUI.php:
```diff
@@ -133,166 +155,71 @@ public function performCommand(string $cmd): void
+    /**
+     * @deprecated still in use?
+     */
     protected function upload(): void
```
If depricated, we need to throw or trigger error indicating that this is depricated!

ferishili commented on classes/class.ilOpencastPageComponentPluginGUI.php:
```diff
@@ -133,166 +155,71 @@ public function performCommand(string $cmd): void
-        $form->setFormAction($this->dic->ctrl()->getFormAction($this));
-
-        return $form;
+        // must be after to avoid changed URLs
```
Is this really needed here?

ferishili commented on classes/class.ilOpencastPageComponentPluginGUI.php:
```diff
@@ -303,224 +230,136 @@ public function create(): void
-        $this->dic->ctrl()->redirect($this, self::CMD_INSERT);
+        $upload_button = ilLinkButton::getInstance();
+        $upload_button->setPrimary(true);
+        //        $this->dic->ctrl()->saveParameter($this, 'rtoken'); // TODO why???
```
Again, please remove this or format the TODO: properly so that IDE plugins could read them!
```suggestion
        //        $this->dic->ctrl()->saveParameter($this, 'rtoken'); // TODO: why???
```

but I don't see any reason to keep it, so better to remove it!

chfsx replied:
Will remove this as well

ferishili reviewed (changes requested):
Thanks @chfsx,
I've submitted my review along with some requested changes.
Special thanks for the redesign with separated classes that have clear goals and responsibilities.

chfsx commented:
Thanks @ferishili, I implemented the requested changes except the translations. @dagraf can you have a look at the suggested changes from @ferishili in the language files? I aks you since some of them follow the mockups you made, therefore you should decide if the changes are OK.

dagraf commented:
The suggested changes from @ferishili in the language files I found are for the explanation belongig to the option "Preview image linked to video". These suggestion clearly improve the explanations. Thx, @ferishili. Therefore and @chfsx: Please accept these changes. 

Are there additional suggetions I missed?

chfsx commented:
Tahsnka for the feedback, I committed those changes as well. FMPOV all requested changes are now done, @ferishili you may have a look at it again. if there's nothing more to change, I could merge this PR (and it's sister PR) this afternoon.

ferishili commented:
> all requested changes are now done

It appears that some change requests have not been seen, I tagged you by "What about here?" text, so that you can keep track!
As soon as they are done, I will proceed with re-checking. Thanks.

chfsx commented:
Thanks @ferishili , very strange, I has those changes before but it seems as if I reverted them on the way... I u0dated the PR again

dagraf commented:
@ferishili: Is it up on you to have a last look on the newest changes of @chfsx? I would like to have this PR merged as soon as possible.

ferishili commented:
I am on it

ferishili reviewed (approved):
Thanks @chfsx
LGTM

@dagraf, it is highly recommended to conduct the tests again!

ferishili commented:
If ReviewApp is in use for the testing, it needs to be updated!

dagraf commented:
@chfsx: I would lik to use the ReviewApp to test the changes made. Can you please update it?

chfsx commented:
The ReviewApp is updating now: https://github.com/opencast-ilias/ReviewApps/actions/runs/15828347040

dagraf commented:
@chfsx: See bugs below.


- [x] When trying to open a page where an Opencast video is inserted, the following error is shown: "Unclosed '{' on line 69 does not match ')'" (see screenshot)

<img width="1246" alt="Bildschirmfoto 2025-06-23 um 22 34 20" src="https://github.com/user-attachments/assets/01d8d29d-e957-4fdb-8e61-af486d0f0719" />

- [x] Filters do not work consistently. Sometimes I get sent back to the page with the "Edit Mode" activated after having clicked on "Apply" or "Reset". Maybe it is connected to which "+" icon I hit to insert a new Opencast videos. When I hit the upper icon, the bug appears. When I hit the lower icon, filter do work as expected. Additional finding: When I insert an Opencast video an then hit on "Replace" an then use the filters, they work as expected.
  
<img width="859" alt="Bildschirmfoto 2025-06-23 um 23 07 24" src="https://github.com/user-attachments/assets/ad4f274f-c29b-4f0f-9bcb-6cc36a7ffdcd" />

ferishili commented on src/Views/Display.php:
```diff
@@ -71,7 +71,7 @@ public function get(): Component|array
         $max_width = $max_width !== '' ? $max_width . 'px' : 'auto';
 
-        $configured_aspect_ratio =  ((float) $this->properties[\ilOpencastPageComponentPluginGUI::PROP_ASPECT_RATIO]) ?? Edit::RATIO_AS_PUBLICATION);
+        $configured_aspect_ratio = (float) (($this->properties[\ilOpencastPageComponentPluginGUI::PROP_ASPECT_RATIO]) ?? Edit::RATIO_AS_PUBLICATION);
```
With this change, we went back to the initial issue of having the `$configured_aspect_ratio` always as float!
This will affect the next comparison of `===`.
So this needs to be:
```suggestion
        $configured_aspect_ratio = ((float) $this->properties[\ilOpencastPageComponentPluginGUI::PROP_ASPECT_RATIO]) ?? Edit::RATIO_AS_PUBLICATION;

```

chfsx replied:
the suggested change is invalid and reintroduces the missing unclosed { error. I will update the commit in another  way

ferishili replied:
Since it is a suggestion, therefore, please adjust it in the IDE, the main goal is to highlight the outcome!

ferishili replied:
resolved in https://github.com/opencast-ilias/OpencastPageComponent/pull/38/commits/3973ce681155335028c8cfdaf0c9f6264642b697

ferishili reviewed (approved):
Thanks @chfsx

chfsx commented:
@dagraf I think I found a solutions for the issue with the filters mentioned above. It seems that a core issue is the reson for the behaviour, but I found a way around it. And it's unlikely to be successful to report this as a core issue, because the plugin does a lot of things that are not explicitly supported in the plugin slot.

dagraf commented:
I can confirm the fix of the last two above mentioned bugs.

@chfsx: Can you merge this PR, or is another review by @ferishili needed because of the fix to the filter?

chfsx commented:
@ferishili this is the only new commit: https://github.com/opencast-ilias/OpencastPageComponent/pull/38/commits/6e62b33ba88e86a22a9a3b25a09640a664a73727

ferishili commented:
Since it is working, I have nothing to add here!

chfsx commented:
mreged locally and release in version 9.1.0
